flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -1753,19 +1751,10 @@ struct EmboxProcOpConversion : public FIROpConversion<fir::EmboxProcOp> {
   using FIROpConversion::FIROpConversion;
 
   mlir::LogicalResult
-  matchAndRewrite(fir::EmboxProcOp emboxproc, OperandTy operands,
+  matchAndRewrite(fir::EmboxProcOp emboxproc, OpAdaptor adaptor,
                   mlir::ConversionPatternRewriter &rewriter) const override {
-    auto loc = emboxproc.getLoc();
-    auto *ctx = emboxproc.getContext();
-    auto ty = convertType(emboxproc.getType());
-    auto c0 = mlir::ArrayAttr::get(ctx, rewriter.getI32IntegerAttr(0));
-    auto c1 = mlir::ArrayAttr::get(ctx, rewriter.getI32IntegerAttr(1));
-    auto un = rewriter.create<mlir::LLVM::UndefOp>(loc, ty);
-    auto r = rewriter.create<mlir::LLVM::InsertValueOp>(loc, ty, un,
-                                                        operands[0], c0);
-    rewriter.replaceOpWithNewOp<mlir::LLVM::InsertValueOp>(emboxproc, ty, r,
-                                                           operands[1], c1);
-    return success();
+    TODO(emboxproc.getLoc(), "fir.emboxproc codegen");
+    return failure();
```

Comment:
A value of type `boxproc` is a pair of pointers. It looks like the removed lowering was close if not correct.

Calling a procedure via an `boxproc` value requires lowering to create a thunk.